tests/tabs/registrar/components/AutoLoginComponent.test.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import mockStore from '../../../config/mockStore';
+import en from '../../../../src/languages/en.json';
+
+import AutoLoginComponent from '../../../../src/app/tabs/registrar/components/AutoLoginComponent';
```

Comment:
Why don't we put tests beside the component? I think scaffolding will be easier to maintain this way.

Reply:
Great question. I went back and forth on this and could go either way. I came up with putting them in their own directory because of these two reasons:

1: Thinking long term, when we have 100% coverage, this is a lot of files that could potentially clutter up the src directory and make it harder for a new developer to come in and understand what is going on.

2: The linter assumes that anything that is in the `src` directory is for production and requires all of the dependencies to be in the `dependencies` array in production. Here, if we put the files in a test directory, the linter will allow all of the testing dependencies to be in the `devDependencies` array.

Reply:
1. But a newcomer would never know where to include unit test for his implementation...
2. Maybe we can create rule for that type of files.

I think both approaches are ok anyway. Let's get opinion from another dev 😄

Reply:
I know that the Angular approach is pretty similar to what Jesse did, anyway, I prefer to have separated tests folder. Mainly because is not `src`, it tests `src` content...

Reply:
Let’s go for unit tests beside unit being tested, integration will go inside /test. If this becomes a mess we can move them out 😁
Another pro: we will realize what to refactor and how to scaffold refactoring when creating this tests